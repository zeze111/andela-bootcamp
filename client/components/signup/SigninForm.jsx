import React, { Component }  from 'react';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import TextFieldGroup from '../common/textFieldGroup';

class signinForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      errors: {},
      isLoading: false
    }

    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onChange(e) {
    this.setState({ [e.target.name]: e.target.value});
  }

  onSubmit(e) {
    this.setState({ errors: {}, isLoading: true });
    e.preventDefault();
    
    this.props.userSigninRequest(this.state)
    .then(() => { 
      this.setState({ redirect: true });
    })
    .catch((error) => { this.setState({ errors: error.response.data, isLoading: false }) });
  }

  render() {
    const { errors } = this.state;
    const { redirect } = this.state;
    if (redirect) {
      return <Redirect to="/" />;
    }

    return (
      <div>
        <form onSubmit={this.onSubmit} className="col s5 offset-s4"> <br/>
          <TextFieldGroup
          label="Email"
          value={this.state.email}
          onChange={this.onChange}
          id="email"
          type="email"
          name="email"
        />
          <TextFieldGroup
          label="Password"
          value={this.state.password}
          onChange={this.onChange}
          id="pwd"
          type="password"
          name="password"
        />
        <div>
          <input disabled={this.state.isLoading} className="btn grey white-text right-align" type="submit" value="Sign in"/>
        </div> <br/>
        {errors && <span className='red-text' style={{ fontSize: 16 + 'px' }}>
          {errors.message}</span>} 
      </form>
    </div>
    );
  }
}

signinForm.propTypes = {
  userSigninRequest: PropTypes.func.isRequired
}


export default signinForm;
