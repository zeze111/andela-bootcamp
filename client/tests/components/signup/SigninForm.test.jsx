import React from 'react';

import SigninForm from '../../../components/signup/SigninForm';
import store from '../../../store';

let props = {
  signIn: jest.fn(() => Promise.resolve())
}

const setup = () => {
  const wrapper = shallow(<SigninForm {...props} store={store} />);
  return { wrapper }
}

describe('onChange', () => {
  it('should change event', () => {
    const event = {
      target: { name: 'email', value: 'Doe@yahoo.com' },
    };
    const { wrapper } = setup();
    const email = wrapper.find('#signInEmail');
    email.simulate('change', event);
    expect(wrapper.instance().state.email).toBe('Doe@yahoo.com');
  });
});

describe('onSubmit', () => {
  it('be called when form is submitted', () => {
    const { wrapper } = setup();
    const event = {
      preventDefault: jest.fn(),
    };
    wrapper.setState({
      email: 'jane@yahoo.com',
      password: 'janedoe',
      errors: {},
      isLoading: false
    });
    wrapper.instance().onSubmit(event);
    expect(props.signIn.mock.calls.length).toEqual(1);
  });
});

describe('onSubmit', () => {
  it('return an error when form is submitted', () => {
    const event = {
      preventDefault: jest.fn(),
    };
    props.signIn = jest.fn(() => Promise.reject());
    const wrapper = shallow(<SigninForm {...props} store={store} />);

    wrapper.setState({
      email: 'jane@yahoo.com',
      password: 'janed',
    });
    wrapper.instance().onSubmit(event);
    expect(props.signIn.mock.calls.length).toEqual(1);
  });
});

describe('Test for Sign up form', () => {
  it('should render component', () => {

    const wrapper = setup();
    expect(wrapper).toMatchSnapshot()
  });
});
