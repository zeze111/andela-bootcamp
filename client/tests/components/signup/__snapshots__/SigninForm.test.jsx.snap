// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Sign up form should render component 1`] = `
Object {
  "wrapper": <div>
    <form
      className="col s10 m7 l6 push-s1 push-m2 push-l3"
      onSubmit={[Function]}
    >
       
      <br />
      <TextFieldGroup
        active=""
        error={Array []}
        id="signInEmail"
        label="Email"
        name="email"
        onChange={[Function]}
        placeholder=""
        req=""
        type="email"
        value=""
      />
      <TextFieldGroup
        active=""
        error={Array []}
        id="signInPwd"
        label="Password"
        name="password"
        onChange={[Function]}
        placeholder=""
        req=""
        type="password"
        value=""
      />
      <div
        className="right-align"
      >
        <input
          className="btn grey white-text"
          disabled={false}
          type="submit"
          value="Sign in"
        />
      </div>
       
      <br />
      <span
        className="red-text error-text"
      />
    </form>
  </div>,
}
`;
